### PR TITLE
Microbuild

### PR DESCRIPTION
**Pull request by zooba**
Updates build and signing to work with our internal build system.

Significant changes:
* NuGet packages are listed in `PreBuild.ps1` and installed without version numbers in directories
* Forward port of #1186 
* Fixed FXCop issues
* Moved sample projects under Templates folder
* Release builds produce full set of VSIXs, FOPXs and MSIs

### CODE REVIEW

File: Python/Product/Django/Project/DjangoProject.cs [hunk 1]
```diff
@@ -57,6 +58,20 @@ partial class DjangoProject :
         public DjangoProject() {
         }
 
+        public void Dispose() {
```
Comment (DinoV) [inline Python/Product/Django/Project/DjangoProject.cs]: Who actually calls this Dispose method?  It's also weird to see the full Dispose pattern with the bool disposing parameter when this doesn't a finalizer that calls it with Dispose(false).

File: Python/Product/Django/Project/DjangoPropertyPage.cs [hunk 2]
```diff
@@ -30,6 +30,16 @@ class DjangoPropertyPage : CommonPropertyPage {
         public DjangoPropertyPage() {
             _control = new DjangoPropertyPageControl(this);
         }
+
```
Comment (DinoV) [inline Python/Product/Django/Project/DjangoPropertyPage.cs]: Same thing here, who's calling this?  Also why are we introducing protected members into classes with no subclasses?  Neither of these will ever be called with Dispose(false)

File: Python/Product/EnvironmentsList/PipExtensionProvider.cs [hunk 3]
```diff
@@ -72,6 +73,15 @@ public sealed class PipExtensionProvider : IEnvironmentViewExtension {
             _cache = PipPackageCache.GetCache(_index, _indexName);
         }
 
+        [SuppressMessage("Microsoft.Usage", "CA2213:DisposableFieldsShouldBeDisposed", MessageId = "_pipCancel",
```
Comment (DinoV) [inline Python/Product/EnvironmentsList/PipExtensionProvider.cs]: It doesn't look like EnvironmentView ever disposes of it's extensions, so this is never called?

File: Python/Product/Profiling/Profiling/BaseHierarchyNode.cs [hunk 4]
```diff
@@ -148,5 +149,20 @@ abstract class BaseHierarchyNode : IVsUIHierarchy {
                 ev.OnPropertyChanged(itemid, propid, flags);
             }
         }
+
```
Comment (DinoV) [inline Python/Product/Profiling/Profiling/BaseHierarchyNode.cs]: This never gets called either?

File: Python/Product/Uwp/Project/PythonUwpPropertyPage.cs [hunk 5]
```diff
@@ -32,6 +32,16 @@ class PythonUwpPropertyPage : CommonPropertyPage {
         public PythonUwpPropertyPage() {
             _control = new PythonUwpPropertyPageControl(this);
         }
+
+        public void Dispose() {
```
Comment (DinoV) [inline Python/Product/Uwp/Project/PythonUwpPropertyPage.cs]: I'm wondering who calls this one as well, but the protected pattern is also certainly not necessary

File: Python/Setup/PythonToolsInstaller/PythonToolsInstaller.wixproj [hunk 6]
```diff
@@ -77,10 +76,6 @@
       <Name>UwpMsm</Name>
       <Project>{83946E81-7A1B-4D3B-927F-4CD67AC95BE7}</Project>
     </ProjectReference>
-    <ProjectReference Include="$(PTVSPRPath)VsLoggerSetup\VsLogger.wixproj" Condition="'$(IncludeVsLogger)' == 'True'">
```
Comment (DinoV) [inline Python/Setup/PythonToolsInstaller/PythonToolsInstaller.wixproj]: We're removing our private telemetry DLL?  That seems like a step backwards

**Comment by DinoV** [comment]
I'm really just wondering about all these random Dispose methods that are introduced that are never getting called.  If we're adding them they should actually be getting called and we probably shouldn't be using the protected Dispose pattern in places where we don't need it which is most of them.

File: Python/Setup/PythonToolsInstaller/PythonToolsInstaller.wixproj (hunk 6 above)
Comment (zooba) [inline Python/Setup/PythonToolsInstaller/PythonToolsInstaller.wixproj]: It's using the wrong system, so we're not getting any data from it anyway. When we figure out how to use the new one then we'll add that in.

**Comment by zooba** [comment]
The Dispose methods are added to keep FXCop happy. I can make them not use the full pattern, but that requires sealing all of the classes which is a breaking change (though we're accepting breaking changes for this release, so I guess that's fine - just automatically went for the non-breaking change).

**Pull request merged**